chromium_src/tools/grit/grit/tool/build.py

```diff
@@ -38,14 +38,18 @@ def _FormatWebuiHeader(root, *_args, **_kwargs):
 
     yield ('// This file is automatically generated by GRIT. Do not edit.\n\n'
            '#pragma once\n\n'
-           '#include <array>\n\n')
+           '#include <array>\n'
+           '#include <string_view>\n\n')
 
     if root.IsAllowlistSupportEnabled():
         yield '#include "ui/base/resource/allowlist.h"\n'
     yield '#include "ui/base/webui/web_ui_util.h"\n\n'
 
     yield 'namespace webui {\n'
 
+    yield ('\ntemplate <int IDS>\n'
+           'inline constexpr std::string_view IDSToName();\n')
```

Comment:
I don't know if we can put this in the `webui` namespace - won't this result in conflicts if we generate strings from multiple `grds` (or is it fine because this is a template?).

Reply:
why do we need these changes for containers?

Reply:
In https://github.com/brave/brave-core/pull/29172 we introduced an extension for Grit to automatically generate WebUI bindings for strings. Its pretty new so I think we're still playing around with how to do different things. This is the first time its been used with upstream WebUIs for example.

The reason this has been introduced is so @goodov can override a string with a formatted version of itself:
https://github.com/brave/brave-core/pull/29230#discussion_r2151023612

I think my preference would be to avoid the need for `IDSToName` by giving the string a different name and not using our Typescript enum to look up the key, though thinking about it, we might need something like this for PluralStrings.

Reply:
> I don't know if we can put this in the `webui` namespace - won't this result in conflicts if we generate strings from multiple `grds` (or is it fine because this is a template?).

It won't conflict with other headers.

> I think my preference would be to avoid the need for IDSToName by giving the string a different name and not using our Typescript enum to look up the key, though thinking about it, we might need something like this for PluralStrings.

I'd like to avoid giving strings a name by hand entirely, but I like the idea of generating this string on TS side by passing `containersLearnMoreUrl` from C++ side.

Let me remove IDSToName for now and go for another approach.